File: cpp/477_汉明距离总和_meidum.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        for (int i=0; i<30; i++) {
            int cnt = 0;
            for (auto num : nums)
                if ((num >> i) & 1)
                    cnt++;
            ans += cnt * (n - cnt);
        }
        return ans;
    }
};

File: python/477_汉明距离总和_meidum.py
class Solution:
    def totalHammingDistance(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        for i in range(30):
            cnt = 0
            for num in nums:
                if (num >> i) & 1:
                    cnt += 1
            ans += cnt * (n - cnt)
        return ans